doc/api/process.md

```diff
@@ -1917,6 +1917,45 @@ console.log('After:', getActiveResourcesInfo());
 //   After: [ 'TTYWrap', 'TTYWrap', 'TTYWrap', 'Timeout' ]
 ```
 
+## `process.getBuiltinModule(id)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `id` {string} ID of the built-in module being requested.
+* Returns: {Object|undefined}
+
+`process.getBuiltinModule(id)` provides a way to load built-in modules
+in a globally available function. ES Modules that need to support
+other environments can use it to conditionally load a Node.js built-in
+when it is run in Node.js, without having to deal with the resolution
+error that can be thrown by `import` in a non-Node.js environment or
+having to use dynamic `import()` which either turns the module into
+an asynchronous module, or turns a synchronous API into an asynchronous one.
+
+```mjs
+if (globalThis.process.getBuiltinModule) {
+  // Run in Node.js, use the Node.js fs module.
+  const fs = globalThis.process.getBuiltinModule('fs');
+  // If `require()` is needed to load user-modules, use createRequire()
+  const module = globalThis.process.getBuiltinModule('module');
+  const require = module.createRequire(import.meta.url);
+  const foo = require('foo');
+}
+```
+
+If `id` specifies a built-in module available in the current Node.js process,
+`process.getBuiltinModule(id)` method returns the corresponding built-in
+module. If `id` does not correspond to any built-in module, `undefined`
+is returned.
+
+`process.getBuiltinModule(id)` accept built-in module IDs that are recognized
+by [`module.isBuiltin(id)`][]. Some built-in modules must be loaded with the
+`node:` prefix, see [built-in modules with mandatory `node:` prefix][].
+The built-in modules returned by `process.getBuiltinModule(id)` are always the
+original modules - that is, it's not affected by [`require.cache`][].
```

Comment:
What does the statement "not affected by `require.cache`" suggest? AFAICT, the returned module can still be modified and may not be in a pristine state.

Reply:
Someone correct me if I'm wrong, but I believe it's possible for one to put a module directly into `require.cache`. So, if you were to set `fs` on it, `process.getBuiltinModule("fs")` will still get you the real one, while `require("fs")` would not. No matter what, patching the builtins still works, though, it's just the resolution from `require` that's affected by `require.cache`.

Reply:
Modified the doc to say:

```md
The references returned by `process.getBuiltinModule(id)` always point to
the built-in module corresponding to `id` even if users modify
[`require.cache`][] in a way for `require(id)` to return something else.
```

Which should be clear enough. In any case it references the documentation of `require.cache` which talks about what that means in detail.